darts/preprocessing/base_transformer.py

```diff
@@ -0,0 +1,32 @@
+
+from typing import TypeVar, Generic, Callable
+from darts.logging import raise_if_not, get_logger
+
+logger = get_logger(__name__)
+T = TypeVar('T')
+
+
+class BaseTransformer(Generic[T]):
+    def __init__(self):
+        self.validators = [
+            lambda x: True
+        ]
+
+    def add_validator(self, validator: Callable[[T], bool]):
+        self.validators.append(validator)
+        return self
+
+    def validate(self, data: T) -> bool:
+        for validator in self.validators:
```

Comment:
If you want you could also go fancy like `raise_if_not(map(lambda v: v(data), self.validators).all(), ...)`

Reply:
Thats fancy :P, hmm I probably shouldn't include implementations details in this proposal but just interface.

Reply:
you're right ;)